deriva/core/ermrest_model.py

```diff
@@ -2240,11 +2262,20 @@ def alter(
 
         return self
 
-    def drop(self):
+    def drop(self, cascade=False):
         """Remove this key from the remote database.
+
+        :param cascade: drop dependent objects.
         """
         if self.name not in self.table.keys.elements:
             raise ValueError('Key %s does not appear to belong to table %s.' % (self, self.table))
+
+        if cascade:
+            for fkey in list(self.table.referenced_by):
+                assert self.table == fkey.pk_table, "Expected key.table and foreign_key.pk_table to match"
+                if self.unique_columns == fkey.referenced_columns:
```

Comment:
I think this needs to be a set-based test, while this `==` operation is on arrays and so won't match if the referenced columns are in a different order

Reply:
Right. Good catch.

Reply:
add set() constructors to turn it into unordered comparison